tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/ConfigBasedTool.cs

```diff
@@ -0,0 +1,354 @@
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Helpers;
+using Azure.Sdk.Tools.Cli.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Sdk.Tools.Cli.Tools.Package;
+
+/// <summary>
+/// Base class for tools that need configuration-based operations, process execution, and common response patterns.
+/// Provides shared functionality for configuration management, validation, response creation, and process execution.
+/// </summary>
+public abstract class ConfigBasedTool : MCPTool
+{
+    protected readonly ISpecGenSdkConfigHelper specGenSdkConfigHelper;
+    protected readonly IProcessHelper? processHelper;
+    protected readonly ILogger logger;
+
+    protected ConfigBasedTool(
+        ISpecGenSdkConfigHelper specGenSdkConfigHelper,
+        ILogger logger,
+        IProcessHelper? processHelper = null)
+    {
+        this.specGenSdkConfigHelper = specGenSdkConfigHelper;
+        this.logger = logger;
+        this.processHelper = processHelper;
+    }
+
+    /// <summary>
+    /// Creates a failure response with the specified error message.
+    /// Works with any response type that inherits from CommandResponse.
+    /// </summary>
+    /// <typeparam name="T">The response type that inherits from CommandResponse.</typeparam>
+    /// <param name="message">The error message to include in the response.</param>
+    /// <returns>A response indicating failure.</returns>
+    protected T CreateFailureResponse<T>(string message) where T : CommandResponse, new()
```

Comment:
I don't know if this helper is necessary and if it is maybe it could be constrained to the ChangelogUpdateTool specifically.